django/db/migrations/operations/utils.py

```diff
@@ -0,0 +1,8 @@
+def is_referenced_by_foreign_key(state, model_name_lower, field, field_name):
+    for state_app_label, state_model in state.models:
+        for f in state.models[state_app_label, state_model].fields:
+            if (f[1].related_model and
+                    '%s.%s' % (state_app_label, model_name_lower) == f[1].related_model.lower()):
+                if (f[1].to_fields[0] is None and field.primary_key) or field_name in f[1].to_fields:
+                    return True
+    return False
```

Comment:
It's a bit of a shame that we have to resort to full scan of the project state but I'm afraid it's the only solution to detect `to_field` references without rendering the model states.